### PR TITLE
User can view a list of courses (Main screen)

### PR DESCRIPTION
**Pull request by mukeshm**
PR for issue #2

### CODE REVIEW

File: src/components/CreateCourse.js [hunk 1]
```diff
@@ -30,7 +31,7 @@ export default class CreateCourse extends Component {
   }
 
   isNameValid(name){
```
Comment (deobald) [inline src/components/CreateCourse.js]: Try to keep validation in its own namespace/module/class/etc, regardless of language.

File: src/components/CreateCourse.js [hunk 2]
```diff
@@ -4,6 +4,7 @@ import { hashHistory } from 'react-router'
 
 ipcRenderer.on('add-course', (event, arg) => {
   console.log('event,arg', arg)
```
Comment (deobald) [inline src/components/CreateCourse.js]: log output might be confusing (mentions `event,arg` but only displays `arg`)

I'm not familiar enough with JS -- is it common to do `console.log` in lieu of a real logging framework? This stuff would be much easier to debug/trace if it were written to file.

File: src/components/ViewCourses.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            <div className='tableColumn courseDate'>
+              Start Date
+            </div>
+            <div className='tableColumn courseDuration'>
+              Duration
+            </div>
+          </div>
+          <div className='tableBody'>
+               {this.state.courseList.map(course=> {
+                  let date = new Date(course.startDate)
```
Comment (deobald) [inline src/components/ViewCourses.js]: Is there a way to pull this variable out of the JSX? Preferably, all templating would just be HTML and tokens.

File: src/components/ViewCourses.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  }
+  render () {
+    return (
+      <div className='container'>
+        <div className='header'>
+          <button className='buttons nav-buttons' onClick={() => hashHistory.push('/')}>
+            Home
+          </button>
+        </div>
+        {this.state.courseList ? this.renderData() : this.renderLoading()}
```
Comment (deobald) [inline src/components/ViewCourses.js]: Same as comment above: Can you pull this logic out of the JSX?

A good thumb rule to keep in mind is: StringTemplate (one of the original templating libs for HTML templating) at one point did not even *contain* a conditional statement, so testing booleans like this was actually impossible to do in the view. This was the right choice on the part of the StringTemplate designers and can still be followed today.

File: src/css/style.css [hunk 5]
```diff
@@ -97,6 +101,69 @@ html, body{
...
+    height: 400px;
+    overflow-y: scroll;
+    display: inline-block;
+    margin-top: 10px;
+}
+
+.tableRow{
+        display: inline-block;
+        font-size: 18px;
+        cursor: pointer;
```
Comment (deobald) [inline src/css/style.css]: Tabs, spaces, x2, x4, x8 ...doesn't matter but remember to keep it consistent across the entire project (and definitely within one file).

I won't comment on the quality of the CSS, sorry... takes too long to read and understand to the level where I can say something meaningful. 😛

**Review by deobald — CHANGES_REQUESTED** [review]
Looks good, @mukeshm ! Just a couple changes and you can merge.

File: src/components/CreateCourse.js (hunk 2 above)
Comment (rsbh) [reply src/components/CreateCourse.js]: Removed the console.log, it was just for development.

File: src/components/ViewCourses.js (hunk 3 above)
Comment (rsbh) [reply src/components/ViewCourses.js]: Created a new method for map function inside component class, can't pull `date` variable outside  the JSX. It is there just to parse start date ISO string into date object

File: src/components/ViewCourses.js (hunk 4 above)
Comment (rsbh) [reply src/components/ViewCourses.js]: moved this conditional statement from render to separate `getData` function.

File: src/components/CreateCourse.js (hunk 2 above)
Comment (mukeshm) [reply src/components/CreateCourse.js]: Will add system-wide logger in #11

File: src/components/CreateCourse.js (hunk 1 above)
Comment (mukeshm) [reply src/components/CreateCourse.js]: Validation is now in a separate module.

**Pull request merged**